src/kbmod/data_interface.py

```diff
@@ -3,12 +3,111 @@
 from astropy.io import fits
 from astropy.wcs import WCS
 import numpy as np
-
-import kbmod.search as kb
+from pathlib import Path
 
 from kbmod.configuration import SearchConfiguration
 from kbmod.file_utils import *
-from kbmod.work_unit import WorkUnit
+from kbmod.search import (
+    ImageStack,
+    LayeredImage,
+    PSF,
+    RawImage,
+)
+from kbmod.wcs_utils import append_wcs_to_hdu_header
+from kbmod.work_unit import WorkUnit, raw_image_to_hdu
+
+
+def load_deccam_layered_image(filename, psf):
+    """Load a layered image from the legacy deccam format.
+
+    Parameters
+    ----------
+    filename : `str`
+        The name of the file to load.
+    psf : `PSF`
+        The PSF to use for the image.
+
+    Returns
+    -------
+    img : `LayeredImage`
+        The loaded image.
+
+    Raises
+    ------
+    Raises a ``FileNotFoundError`` if the file does not exist.
+    Raises a ``ValueError`` if any of the validation checks fail.
+    """
+    if not Path(filename).is_file():
+        raise FileNotFoundError(f"{filename} not found")
+
+    img = None
+    with fits.open(filename) as hdul:
+        if len(hdul) < 4:
+            raise ValueError("Not enough extensions for legacy deccam format")
+
+        # Extract the obstime trying from a few keys and a few extensions.
+        obstime = 0.0
```

Comment:
1) Is it ever possible for the obstime to be missing? Would we want to fail if so?

2) Currently we set the default obstime to -1 and check for that here (https://github.com/dirac-institute/kbmod/blob/1cc181adfba39c29d8ebc3c76f3e9746409622b7/src/kbmod/data_interface.py#L217),  so we should probably still be using `obstime = -1.0`

Reply:
For 1: yes. There were legacy files without times where we needed to read obstime from an auxiliary text file. So we don't want it to fail outright in this function.
For 2: Good call. I changed it.